src/data/datasets/representation/each_to_each_retrieval_dataset.py

```diff
@@ -0,0 +1,129 @@
+from typing import Union, Optional
+
+import torch
+import pandas as pd
+from albumentations import BasicTransform
+from albumentations.core.composition import BaseCompose
+
+
+from src.data.datasets.base import ImageDataset
+
+
```

Comment:
Create one class with defaults `gallery_*` arguments set to empty values. In the default case each query will be searched for relevant while the gallery is a set of the remaing items (both queries and relevant)